azure/__init__.py

```diff
@@ -1018,6 +614,299 @@ def _lower(text):
     return text.lower()
 
 
+class _ETreeXmlToObject(object):
+    @staticmethod
+    def parse_response(response, return_type):
+        '''
+        Parse the HTTPResponse's body and fill all the data into a class of
+        return_type.
+        '''
+        root = ETree.fromstring(response.body)
+        xml_name = return_type._xml_name if hasattr(return_type, '_xml_name') else return_type.__name__ 
+        if root.tag == xml_name:
+            return _ETreeXmlToObject._parse_response_body_from_xml_node(root, return_type)
+
+        return None
+
+
+    @staticmethod
+    def parse_enum_results_list(response, return_type, resp_type, item_type):
+        """resp_body is the XML we received
+    resp_type is a string, such as Containers,
+    return_type is the type we're constructing, such as ContainerEnumResults
+    item_type is the type object of the item to be created, such as Container
+
+    This function then returns a ContainerEnumResults object with the
+    containers member populated with the results.
+    """
+
+        # parsing something like:
+        # <EnumerationResults ... >
+        #   <Queues>
+        #       <Queue>
+        #           <Something />
+        #           <SomethingElse />
+        #       </Queue>
+        #   </Queues>
+        # </EnumerationResults>
+        return_obj = return_type()
+        root = ETree.fromstring(response.body)
+
+        items = []
+
+        for container_element in root.findall(resp_type):
+            for item_element in container_element.findall(resp_type[:-1]):
+                items.append(_ETreeXmlToObject.fill_instance_element(item_element, item_type))
+
+        for name, value in vars(return_obj).items():
+            # queues, Queues, this is the list its self which we populated
+            # above
+            if name == resp_type.lower():
+                # the list its self.
+                continue
+            value = _ETreeXmlToObject.fill_data_member(root, name, value)
+            if value is not None:
+                setattr(return_obj, name, value)
+
+        setattr(return_obj, resp_type.lower(), items)
+        return return_obj
+
+
+    @staticmethod
+    def parse_simple_list(response, type, item_type, list_name):
+        respbody = response.body
+        res = type()
```

Comment:
Call this `return_type` rather than `type`? Avoids confusion, since `type` is a built-in class.